src/templates.rs

```diff
@@ -103,125 +107,178 @@ pub fn render_template_file_tree<'a, T: Serialize>(
-
-                if files_to_create.len() > 0 {
-                    let delimiter = "\n----END_OF_FILE_DELIMITER----\n";
-
-                    let each_if_re = Regex::new(
-                    r"(?P<c>(.)*)/\{\{#each (?P<b>([^\{\}])*)\}\}\{\{#if (?P<d>([^\{\}])*)\}\}(?P<a>(.)*)\{\{/if\}\}\{\{/each\}\}.hbs\z",
-                )
-                .unwrap();
-                    let b = re.replace(path.to_str().unwrap(), "${b}");
-                    let new_all_contents = match each_if_re.is_match(path.to_str().unwrap()) {
-                        true => {
-                            let d = each_if_re.replace(path.to_str().unwrap(), "${d}");
-                            format!(
-                                "{{{{#each {} }}}}{{{{#if {} }}}}\n{}{}{{{{/if}}}}{{{{/each}}}}",
-                                b, d, contents, delimiter
-                            )
-                        }
-
-                        false => format!(
-                            "{{{{#each {} }}}}\n{}{}{{{{/each}}}}",
-                            b, contents, delimiter
-                        ),
-                    };
-                    let new_contents = render_template_file(
-                        &h,
-                        existing_app_file_tree,
-                        &path,
-                        &new_all_contents,
-                        &value,
-                    )?;
-                    let new_contents_split: Vec<String> = new_contents
-                        .split(delimiter)
-                        .into_iter()
-                        .map(|s| s.to_string())
-                        .collect();
-
-                    for (i, f) in files_to_create.into_iter().enumerate() {
-                        let target_path = PathBuf::from(path_prefix.clone()).join(f);
-
-                        transformed_templates
-                            .insert(target_path, Some(new_contents_split[i].clone()));
+            let each_regex = Regex::new(EACH_TEMPLATE_REGEX).context("EACH_TEMPLATE_REGEX is invalid")?;
+            let if_regex = Regex::new(IF_TEMPLATE_REGEX).context("IF_TEMPLATE_REGEX is invalid")?;
+
+            match (each_regex.is_match(path_str), if_regex.is_match(path_str)) {
+                (true, _) => handle_each_regex_template(
+                    each_regex,
+                    h,
+                    &path,
+                    path_str,
+                    &contents,
+                    existing_app_file_tree,
+                    data,
+                    &mut transformed_templates,
+                )?,
+                (_, true) => handle_if_template_regex(
+                    if_regex, 
+                    h,
+                    path_str,
+                    &contents,
+                    existing_app_file_tree,
+                    data,
+                    &mut transformed_templates
+                )?,
+                _ => {
+                    if let Some(e) = path.extension() {
```

Comment:
Somewhere above there was a nice refactor to do this with a `map_or`, same here to avoid the nested `if`?

Reply:
Fixed